torch/distributed/_shard/replicated_tensor.py

```diff
@@ -123,3 +123,27 @@ def validate(self) -> bool:
                     f"ReplicatedTensor have different values on rank {current_rank} and {rank}")
 
         return True
+
+class ReplicatedParameter(ReplicatedTensor, torch.nn.Parameter):
```

Comment:
@albanD I couldn't pass in a ReplicatedTensor to `torch.nn.Parameter` and have `isinstance()` still return ReplicatedTensor. As a result, I had to do something messy like this. Not sure if there is a better way to do this.

Reply:
cc @jbschlosser

Reply:
This is a bit delayed due to my medical leave, but wanted to leave an update here for posterity: #73459 is in now - please try it out if you'd like :)